lib/elliptic/ec/signature.js

```diff
@@ -23,35 +23,74 @@ function Signature(options, enc) {
 }
 module.exports = Signature;
 
+function getLength(buf, p) {
+  var initial = buf[p.place++];
+  if (!(initial & 0x80)) {
+    return initial;
+  }
+  var octetLen = initial & 0xf;
+  var byteLength = octetLen;
+  var val = buf[p.place + --byteLength];
+  var mul = 1;
+  while (byteLength > 0 && (mul *= 0x100)) {
+    val += buf[p.place + --byteLength] * mul;
+  }
+  p.place += octetLen;
+  return val;
+}
 Signature.prototype._importDER = function _importDER(data, enc) {
   data = utils.toArray(data, enc);
-  if (data.length < 6 || data[0] !== 0x30 || data[2] !== 0x02)
-    return false;
-  var total = data[1];
-  if (1 + total > data.length)
-    return false;
-  var rlen = data[3];
-  // Short length notation
-  if (rlen >= 0x80)
-    return false;
-  if (4 + rlen + 2 >= data.length)
+  var p = {};
+  p.place = 0;
+  if (data[p.place++] !== 0x30) {
     return false;
-  if (data[4 + rlen] !== 0x02)
+  }
+  getLength(data, p);
+  if (data[p.place++] !== 0x02) {
     return false;
-  var slen = data[5 + rlen];
-  // Short length notation
-  if (slen >= 0x80)
+  }
+  var rlen = getLength(data, p);
+  var r = data.slice(p.place, rlen + p.place);
+  p.place += rlen;
+  if (data[p.place++] !== 0x02) {
     return false;
-  if (4 + rlen + 2 + slen > data.length)
+  }
+  var slen = getLength(data, p);
+  if (data.length !== slen + p.place) {
     return false;
+  }
+  var s = data.slice(p.place, slen + p.place);
+  if (!r[0] && (r[1] & 0x80)) {
+    r = r.slice(1);
+  }
+  if (!s[0] && (s[1] & 0x80)) {
+    s = s.slice(1);
+  }
 
-  this.r = new bn(data.slice(4, 4 + rlen));
-  this.s = new bn(data.slice(4 + rlen + 2, 4 + rlen + 2 + slen));
+  this.r = new bn(r);
+  this.s = new bn(s);
   this.recoveryParam = null;
 
   return true;
 };
 
+function constructLength(arr, len) {
+  if (len < 0x80) {
+    arr.push(len);
+    return;
+  }
+  var octets = 1 + (Math.log(len) / Math.LN2 >> 3);
```

Comment:
I thought that `/` has less priority than `>>>`, so it will do `>>>` first, and divide later. Turns out you are right, and this is correct statement.